src/testing/testing.md

```diff
@@ -0,0 +1,549 @@
+
+#### Example of Specific API Calls:
+
+```typescript
+// tests/customer/customerDetail.spec.ts
+
+import { test } from '@playwright/test';
+import { DataSeederUtil } from '../utils/data.util.ts';
+import { CLAIM_TASK_INDEX, CLAIM_TASK } from '../fixtures/task.fixture.ts';
+import { CONSULTANT_INDEX_1, CONSULTANT_INDEX_2 } from '../fixtures/consultant.fixture.ts';
+
+test.beforeEach(async ({ page }) => {
+  // Setup shared mock data
+  await DataSeederUtil.setup(page);
+  
+  // Mocking specific API calls for tasks and consultants
+  await InterceptorUtil.getPaginated(page, 'tasks/claims', [CLAIM_TASK_INDEX]);
+  await InterceptorUtil.get(page, `tasks/claims/${CLAIM_TASK_INDEX.uuid}`, CLAIM_TASK);
+  await InterceptorUtil.getPaginated(page, 'consultants', [
+    CONSULTANT_INDEX_1,
+    CONSULTANT_INDEX_2,
+  ]);
+  await InterceptorUtil.post(page, `tasks/claims/${CLAIM_TASK_INDEX.uuid}`, CLAIM_TASK);
+});
+
+test('should display customer details correctly', async ({ page }) => {
+  // Test logic for displaying customer details
+});
+```
+
+### When to Use Shared vs. Specific API Calls?
+
+- **Shared API Calls:** Useful when mocking frequently used API calls such as departments, KPIs, and FAQs. This helps minimize code duplication and centralizes common logic.
+- **Specific API Calls:** Used when a test requires unique data that is not reusable by other tests. For example, a test that needs to verify a specific error or edge case.
+
+### Best Practices for API Mocking
+
+1. **Route Interception:**
+    - Use Playwright’s route feature to mock API responses. This ensures that our tests are always consistent, regardless of the external API's status.
+
+2. **Data Builders:**
+    - Use builders like `DepartmentBuilder` and `CustomerBuilder` to create mock data. This makes it easy to generate specific objects with desired properties and promotes reusability.
+
+3. **Parallelization and Isolation:**
+    - By mocking API calls, we can run tests in parallel without relying on a shared backend. This speeds up tests and ensures reliability.
+
+## 10. Builder Pattern Strategy
+
+When implementing the Builder Pattern in our project, we need to consider several key factors. The goal of this pattern is to create objects flexibly and controllably, especially when objects are complex or have many different properties. Below are the conventions and approaches for building a Builder, including how to handle circular dependencies, best practices for data loading, and using other builders within a builder.
+
+### Placement of Builders:
+
+- **Location:** Builders are placed in the same directory as their corresponding model.
+- **Naming Convention:** `{modelName}.builder.ts`.
+- **Example:** If you have a model `customer.model.ts`, place the builder in `customer.builder.ts`.
+
+### Avoiding Circular Dependencies:
```

Comment:
zit overal veel duplicate en heel veel blablablabla in deze delen, veel simpeler schrijven met minder woorden